.pre-commit

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# Check if js or scss is updated
+
+if git diff --cached --name-only --diff-filter=ACM | grep "resources/js/src\|resources/scss"
+then
+    # Check if already compiled
+    if git diff --cached --name-only --diff-filter=ACM | grep "resources/js/dist\|resources/css"
+    then
+        exit 0;
+    fi
+
+    # Changes not compiled
+    gulp js
+    gulp sass
+
+    if git diff --name-only --diff-filter=ACM | grep "resources/js/dist\|resources/css"
+    then
+        printf "ERROR: JS/CSS was not compiled before pushing. Run gulp\n"
```

Comment:
I'd like to have `gulp` in quotation marks. This makes it more obvious that this is a command.

Reply:
The moment you've commited, gulp was already executed. You could also just mention that these files need to be added to the current commit in order to be commited.